source/nodejs/adaptivecards/src/utils.ts

```diff
@@ -25,21 +25,43 @@ export function appendChild(node: Node, child: Node) {
 }
 
 export function setProperty(target: any, propertyName: string, propertyValue: any, defaultValue: any = undefined) {
-	if (propertyValue && (!defaultValue || defaultValue !== propertyValue)) {
+    let targetIsAlreadySet = target[propertyName] !== undefined;
+
+    if (propertyValue === undefined || propertyValue === null) {
+        if (!targetIsAlreadySet) {
+            delete target[propertyName];
+        }
+    }
+    else if (propertyValue !== defaultValue) {
 		target[propertyName] = propertyValue;
 	}
 	else {
-		delete target[propertyName];
+        if (!targetIsAlreadySet) {
+            delete target[propertyName];
+        }
 	}
 }
 
-export function setEnumProperty(enumType: { [s: number]: string }, target: any, propertyName: string, propertyValue: number, defaultValue?: number) {
-	if (defaultValue === undefined || defaultValue !== propertyValue) {
-		target[propertyName] = enumType[propertyValue];
-	}
-	else {
-		delete target[propertyName];
-	}
+export function setEnumProperty(enumType: { [s: number]: string }, target: any, propertyName: string, propertyValue: number, defaultValue: number = undefined) {
+    let targetValue = target[propertyName];
+
+    let canDeleteTarget = targetValue == undefined ? true : enumType[targetValue] !== undefined;
+    
+    if (propertyValue == defaultValue) {
```

Comment:
===

Reply:
There are just so many occurrences of this in the code... I'll eventually fix them all in one shot.